services/solr-names-updater/src/solr_names_updater/worker.py

```diff
@@ -0,0 +1,143 @@
+from queue_common.service_utils import QueueException, logger
+from requests import RequestException
+from sentry_sdk import capture_message
+from sqlalchemy.exc import OperationalError
+from urllib3.exceptions import NewConnectionError
+
+from solr_names_updater import config
+from solr_names_updater.names_processors.names import (
+    process_add_to_solr as process_names_add,
+    process_delete_from_solr as process_names_delete)
+from solr_names_updater.names_processors.possible_conflicts \
+    import (process_add_to_solr as process_possible_conflicts_add,
+            process_delete_from_solr as process_possible_conflicts_delete)
+
+
+qsm = QueueServiceManager()  # pylint: disable=invalid-name
+APP_CONFIG = config.get_named_config(os.getenv('DEPLOYMENT_ENV', 'production'))
+FLASK_APP = Flask(__name__)
+FLASK_APP.config.from_object(APP_CONFIG)
+db.init_app(FLASK_APP)
+
+
+def is_nr_state_change_msg_type(msg: dict):
+    """Check message is of type nr state change."""
+    if msg and msg.get('type', '') == 'bc.registry.nr_state.changed':
+        return True
+
+    return False
+
+
+def is_nr_furnished_change_msg_type(msg: dict):
+    """Check message is of type nr furnished changed."""
+    if msg and msg.get('type', '') == 'bc.registry.nr_furnished.changed':
+        return True
+
+    return False
+
+
+def is_processable(msg: dict):
+    """Determine if message is processable using message type of msg."""
+    if msg and \
+        (is_nr_state_change_msg_type(msg)
+         or is_nr_furnished_change_msg_type(msg)):
+        return True
+
+    return False
+
+
+async def process_nr_event_message(msg: dict, flask_app: Flask):
+    """Update solr accordingly based on incoming nr state changes."""
+    if not flask_app or not msg:
+        raise QueueException('Flask App or msg not available.')
+
+    with flask_app.app_context():
+        logger.debug('entering processing of nr event msg: %s', msg)
+
+        if is_nr_state_change_msg_type(msg):
+            data = msg.get('data')
+            new_state = data.get('newState')
+            if new_state in ('APPROVED', 'CONDITIONAL'):
+                process_names_add(data)
+            elif new_state in ('CANCELLED', 'CONSUMED'):
+                process_names_delete(data)
+                process_possible_conflicts_delete(data)
+            else:
+                logger.debug('no names processing required message %s', msg)
+        # todo need to determine if need another msg type for handling possible conflicts
+        # elif is_nr_furnished_change_msg_type(msg):
+        #     data = msg.get('data')
+        #     nr_state = data.get('nrState')
+        #     is_furnished = data.get('isFurnished')
+        #     if is_furnished and nr_state in ('APPROVED', 'CONDITIONAL'):
+        #         process_possible_conflicts_add(data)
+        #     else:
+        #         logger.debug("no possible conflict processing required message %s", msg)
```

Comment:
in talking with @kialj876 about possible conflicts, it seems like we need to publish a message that indicates that a NR has been furnished to be able to pick up on possible conflicts.  @thorwolpert what are your thoughts on this and if we do this, should we publish this using another message type?

Reply:
Yeah this is what I was thinking

Reply:
@thorwolpert When adding a possible conflict it seems like a NR state change is not sufficient and we need to publish an event that an NR has been furnished(*requests.furnished* field). The queue listener will then push a possible conflict to solr if the furnished NR also has a NR state of approved or conditional. Are you ok with publishing a different message type when a NR is furnished so the solr-names-updater queue listener can pick it up?

Reply:
no, the state change is sufficient.
there is no dependency on furnishing, there is business requirement to not allow the same, or too-similar of a name, so delaying an approved name from showing up as a conflict works heavily against that

Reply:
Removed commented out logic checking for furnished state and added possible conflict immediately when request state change is `APPROVED` or `CONDITIONAL`.